src/components/strategies/create/CreateForm.tsx

```diff
@@ -0,0 +1,167 @@
+import { FC, FormEvent, ReactNode, useEffect } from 'react';
+import { Token } from 'libs/tokens';
+import { createStrategyEvents } from 'services/events/strategyEvents';
+import { useSearch } from '@tanstack/react-router';
+import { TokensOverlap } from 'components/common/tokensOverlap';
+import { Button } from 'components/common/button';
+import { useCreateStrategy } from './useCreateStrategy';
+import { getStatusTextByTxStatus } from '../utils';
+import { useModal } from 'hooks/useModal';
+import { cn } from 'utils/helpers';
+import { m } from 'libs/motion';
+import { items } from 'components/strategies/common/variants';
+import { useWeb3 } from 'libs/web3';
+import { ReactComponent as IconWarning } from 'assets/icons/warning.svg';
+import { BaseOrder } from 'components/strategies/common/types';
+import { StrategyType } from 'libs/routing';
+import style from 'components/strategies/common/form.module.css';
+
+const title: Record<StrategyType, string> = {
+  disposable: 'Disposable',
+  recurring: 'Recurring',
+  overlapping: 'Concentrated Liquidity',
+};
+
+interface HeaderProps {
+  type: StrategyType;
+  base: Token;
+  quote: Token;
+}
+export const CreateFormHeader = (props: HeaderProps) => {
+  const { base, quote, type } = props;
+  return (
+    <m.header
+      variants={items}
+      key="createStrategyBuyTokens"
+      className="rounded-10 bg-background-900 flex flex-col gap-10 p-20"
+    >
+      <div className="flex gap-10">
+        <TokensOverlap tokens={[base, quote]} size={32} />
+        <hgroup>
+          <h2 className="text-14 flex gap-6">
+            <span>{base?.symbol}</span>
+            <span role="separator" className="text-white/60">
+              /
+            </span>
+            <span>{quote?.symbol}</span>
+          </h2>
+          <p className="text-14 capitalize text-white/60">{title[type]}</p>
+        </hgroup>
+      </div>
+      <p className="text-12 font-weight-400 flex items-center text-white/60">
+        <IconWarning className="ml-6 mr-10 w-14 flex-shrink-0" />
+        Rebasing and fee-on-transfer tokens are not supported
+      </p>
+    </m.header>
+  );
+};
+
+interface FormProps {
+  type: StrategyType;
+  base: Token;
+  quote: Token;
+  order0: BaseOrder;
+  order1: BaseOrder;
+  approvalText?: string;
+  children: ReactNode;
+}
+
+export const CreateForm: FC<FormProps> = (props) => {
+  const { base, quote, order0, order1, type, children } = props;
+  const { openModal } = useModal();
+  const { user } = useWeb3();
+  const search = useSearch({ strict: false }) as any;
```

Comment:
Can you set the type of the search here?

Reply:
Not here unfortunately since it's used on multiple pages